st2api/st2api/controllers/v1/webhooks.py

```diff
@@ -154,10 +158,15 @@ def post(self, hook, webhook_body_api, headers, requester_user):
 
             payload['headers'] = headers
             payload['body'] = body
+
             # Dispatch trigger instance for each of the trigger found
-            for trigger in triggers:
-                self._trigger_dispatcher.dispatch(trigger, payload=payload,
-                    trace_context=trace_context)
+            for trigger_dict in triggers:
+                # TODO: Instead of dispatching the whole dict we should just
+                # dispatch TriggerDB.ref or similar
+                self._trigger_dispatcher_service.dispatch_with_context(trigger=trigger_dict,
```

Comment:
nit pick: move `trigger=trigger_dict` to a newline so it's consistent with styling on the other parameters below.

Reply:
Yeah, I actually hate the dict stuff - didn't even know we support it here (since other dispatch function doesn't really document it can also be a dict) - I assumed it always need to be either a trigger_type_ref or trigger_ref.

Sadly that's not the case, but something we should change and improve in the future.